### ISSUE TITLE
VST SDK path

### ISSUE THREAD

**Issue by Taikakim** [phantom-code/airwave#92]
OK, really basic problem, the compiler commands don't find the SDK.
I installed it in ~/VST3/SDK (where resides VST_SDK folder that came from Steinberg's zip)

When I try: cmake -DCMAKE_BUILD_TYPE="Release" -DCMAKE_INSTALL_PREFIX=/opt/airwave -DVSTSDK_PATH=${HOME}/VST3/SDK ..

I get the  "VST SDK is not found" error.

I tried moving the contents of VST_SDK one level down to ~/VST3/SDK, but that did not help.

**Comment by Taikakim** [comment]
I found the mistake: Steinberg has changed the directory structure.

You need to copy the two directories under the VST2 sub-directory to the directory where the VSTSDK_PATH is set. So one level down, basically.

**Comment by Taikakim** [comment]
But installation went somewhere strange, I got:
Install the project...
-- Install configuration: "Release"
-- Installing: /downloads/airwave-master/bin/airwave-plugin.so
-- Installing: /downloads/airwave-master/bin/airwave-host-64.exe
-- Installing: /downloads/airwave-master/bin/airwave-host-64.exe.so
-- Installing: /downloads/airwave-master/bin/airwave-host-32.exe
-- Installing: /downloads/airwave-master/bin/airwave-host-32.exe.so
-- Installing: /downloads/airwave-master/bin/airwave-manager
-- Installing: /downloads/airwave-master/share/icons/hicolor/48x48/apps/airwave-manager.png
-- Installing: /downloads/airwave-master/share/applications/airwave-manager.desktop

That is the root directory of my system where those directories where created. Now I'm not sure where to copy all that. To /usr/bin? How about the .so plugins?

**Comment by ProdByJuliot** [comment]
So, for creating the VST/SDK directory in the first place, can you just manually make a folder, or is there a terminal process needed? I'm kind of a noob when it comes to manually installing programs on linux.

So, could I just make a new folder in home and name it VST/SDK? Then, extract the Steinberg tools, and take them out of the main folder?

**Comment by Taikakim** [comment]
Hmm I can't compile. When I make, I get until some point when I get this error:
ld: Relocatable linking with relocations from format elf64-x86-64 (/opt/wine-devel/lib64/wine/libwinecrt0.a(exe_entry.o)) to format elf32-i386 (airwave-host-32.WqklSw.o) is not supported

**Comment by gordol** [comment]
https://web.archive.org/web/*/https://download.steinberg.net/sdk_downloads/vstsdk369_01_03_2018_build_132.zip

**Comment by tosho** [comment]
do I need the SDK if I've installed airwave with .deb ?

**Comment by seanjean3000** [comment]
I've tried dozens of different paths and also moving folders around in the SDK from Steinberg, but no matter what I do I keep getting an error message that the SDK cannot be found or that the specified directory doesn't contain CMakeLists.txt (which I plainly see is there). Can any kind soul help me get this sorted?

**Comment by seanjean3000** [comment]
Hey thanks. Still no dice; downloaded the archived version of the SDK and tried putting it in all sorts of different places / changing the path, but nothing. Still getting an error that the SDK can't be found. Probably just making some basic mistake but I give up.

**Comment by vixl152** [comment]
Hello together,

I had the same error when installing via cmake:

> When I try: cmake -DCMAKE_BUILD_TYPE="Release" -DCMAKE_INSTALL_PREFIX=/opt/airwave -DVSTSDK_PATH=${HOME}/VST3_SDK ..
> 
> I get the "VST SDK is not found" error.

What worked for me was using the SDK from gordols archive. 

> https://web.archive.org/web/*/https://download.steinberg.net/sdk_downloads/vstsdk369_01_03_2018_build_132.zip

Also I had to point to the VST2_SDK directory! Like this:

`cmake -DCMAKE_BUILD_TYPE="Release" -DCMAKE_INSTALL_PREFIX=/home/me/opt/airwave -DVSTSDK_PATH=/home/me/Downloads/VST_SDK/VST2_SDK ..`

But know I'm getting the an error doing the make:

[ 20%] Building CXX object src/host/CMakeFiles/airwave-host-32.dir/__/common/dataport.cpp.o
In file included from /usr/include/c++/8/ext/string_conversions.h:41,
                 from /usr/include/c++/8/bits/basic_string.h:6400,
                 from /usr/include/c++/8/string:52,
                 from /usr/include/c++/8/stdexcept:39,
                 from /usr/include/c++/8/array:39,
                 from /usr/include/c++/8/tuple:39,
                 from /usr/include/c++/8/bits/unique_ptr.h:37,
                 from /usr/include/c++/8/memory:80,
                 from /home/viktor/Downloads/airwave/src/common/types.h:6,
                 from /home/viktor/Downloads/airwave/src/common/dataport.h:5,
                 from /home/viktor/Downloads/airwave/src/common/dataport.cpp:1:

**/usr/include/c++/8/cstdlib:75:15: Fatal Error stdlib.h: File or directory not found
 #include_next <stdlib.h>**

Someone any idea, how to solve this one? 

I'm sitting on a Fedora 29 machine.

**Comment by gordol** [comment]
@vixl152 have you installed the libc-dev packages?

**Comment by vixl152** [comment]
@gordol: Yes I have. Did the yum install like in the install instructions:

`sudo yum -y install gcc-c++ git cmake wine wine-devel wine-devel.i686 file file-devel libX11-devel libX11-devel.i686 qt5-devel glibc-devel.i686 glibc-devel`

Here on my machine:

```
[vixl152@fedora build]$ rpm -qa|grep glibc-devel

glibc-devel-2.28-26.fc29.x86_64
glibc-devel-2.28-26.fc29.i686
```

**Comment by mihahauke** [comment]
@seanjean3000
I don't think that you are doing anything wrong. Cmake requires aeffect.h and aeffectx.h which are not in the new SDK cause fuck legacy code and VST 2. It looks like some parts of code need to be rewriten cause 5 minute code substitutions I made weren't enough and caused furter errors.

I might be wrong, I'm not much into C/C++. Perhaps it's just about imports but perhaps not . . . sad face, sad face, wink wink

**Comment by gordol** [comment]
y'all, using the archive.org link to grab the old SDK works fine.

![screenshot](https://node.zeneval.com/snaps/4a17857855203cbb6c629a9d70a1631f.png)

for vst3, native linux is supported, almost globally across every VST that I've tried.

**Comment by snate28** [comment]
> Hello together,
> 
> I had the same error when installing via cmake:
> 
> > When I try: cmake -DCMAKE_BUILD_TYPE="Release" -DCMAKE_INSTALL_PREFIX=/opt/airwave -DVSTSDK_PATH=${HOME}/VST3_SDK ..
> > I get the "VST SDK is not found" error.
> 
> What worked for me was using the SDK from gordols archive.
> 
> > https://web.archive.org/web/*/https://download.steinberg.net/sdk_downloads/vstsdk369_01_03_2018_build_132.zip
> 
> Also I had to point to the VST2_SDK directory! Like this:
> 
> `cmake -DCMAKE_BUILD_TYPE="Release" -DCMAKE_INSTALL_PREFIX=/home/me/opt/airwave -DVSTSDK_PATH=/home/me/Downloads/VST_SDK/VST2_SDK ..`
> 
> But know I'm getting the an error doing the make:
> 
> [ 20%] Building CXX object src/host/CMakeFiles/airwave-host-32.dir/__/common/dataport.cpp.o
> In file included from /usr/include/c++/8/ext/string_conversions.h:41,
> from /usr/include/c++/8/bits/basic_string.h:6400,
> from /usr/include/c++/8/string:52,
> from /usr/include/c++/8/stdexcept:39,
> from /usr/include/c++/8/array:39,
> from /usr/include/c++/8/tuple:39,
> from /usr/include/c++/8/bits/unique_ptr.h:37,
> from /usr/include/c++/8/memory:80,
> from /home/viktor/Downloads/airwave/src/common/types.h:6,
> from /home/viktor/Downloads/airwave/src/common/dataport.h:5,
> from /home/viktor/Downloads/airwave/src/common/dataport.cpp:1:
> 
> **/usr/include/c++/8/cstdlib:75:15: Fatal Error stdlib.h: File or directory not found #include_next <stdlib.h>**
> 
> Someone any idea, how to solve this one?
> 
> I'm sitting on a Fedora 29 machine.

If you did solve the problem, can you please share?

**Comment by gordol** [comment]
you need the libc dev packages. some distros package them in a wrapper called `build-essential`, but it may be called `libc-dev` or `libc6-dev`.

**Comment by Frenkfil** [comment]
For new SDK package change in CMakeLists.txt to this:
```
 find_path(VSTSDK_INCLUDE_DIR NAMES audioeffect.h audioeffectx.h
        PATHS "${VSTSDK_PATH}/VST2_SDK/public.sdk/source/vst2.x")

if(NOT VSTSDK_INCLUDE_DIR)
        message(FATAL_ERROR "VST SDK is not found. You should set the VSTSDK_PATH variable "
                        "to the directory, where your copy of the VST SDK is located.")
endif()
``` :)

**Comment by QkiZMR** [comment]
Compiling with the newest version of SDK is unsuccessful. Can someone share a link to last working version of SDK?

**Comment by ndri** [comment]
The SDK from https://web.archive.org/web/20180201000000*/https://download.steinberg.net/sdk_downloads/vstsdk369_01_03_2018_build_132.zip still works. Just get the one from 2018, not 2020.

**Comment by QkiZMR** [comment]
Sometimes works but transfer is interrupted by server. After 6 or 7 trials I downloaded file.

**Comment by ljleb** [comment]
Right now, the link seems dead. Is there any other way to get a 2018 release? Or should we update the code to use the new VST SDK 3?

**Comment by ndri** [comment]
> Right now, the link seems dead. Is there any other way to get a 2018 release? Or should we update the code to use the new VST SDK 3?

I still have the zip and have uploaded it to Drive for you: https://drive.google.com/file/d/15sOCzi7TB63YRdYTOK7rGv2KPQ4wXT9C/view?usp=sharing

**Comment by alexxmed** [comment]
I have tried all of these suggestions, and frankly, things are a mess now. I wonder if anyone can provide a simple 1, 2, 3 type guide that actually works. For example, by "Go to the airwave source directory and execute the following commands:" do you mean:
cd airwave/src 
or 
cd airwave
Should airwave be extracted from within VST_SDK or side by side with it so that both VST_SDK and airwave are under the home directory (~/VST_SDK and ~/airwave)?
Why does it not seem to matter where I put things, I still get the error that "VST SDK is not found" .
So close and yet, no-where.

**Comment by NARFNra** [comment]
I ran into this error while trying to compile. I fixed it by downloading the legacy version of VST_SDK from ndri's reply.

After that, I checked the cmake file for airwave. It's looking for this specific directory.

`find_path(VSTSDK_INCLUDE_DIR NAMES aeffect.h aeffectx.h
	PATHS "${VSTSDK_PATH}/pluginterfaces/vst2.x/")`

I searched and pluginterfaces/vst2.x is directly under VST2_SDK in the unpacked folder, so the location you need to point to for airwave to build is...

`-DVSTSDK_PATH=[Wherever-You-Extracted-It-To]/VST_SDK/VST2_SDK`

After I did this, simply running the cmake line pointing at the location of airwave's cmake worked.

Unfortunately, I later ran into a separate error involving my wine version, but I believe that it's unrelated to this problem.
